### ISSUE TITLE
can't get the correct result

### ISSUE THREAD

**Issue by jenyichen** [Kriechi/icp20100-barometric-pressure-sensor-arduino#1]
Hi,
I can't get the correct result with the ESP32 NodeMCU-32S board.  
I wonder if it could be a hardware problem or a software problem? Thank you very much. 
The output is as follows:
15:23:50.521 -> ICP20100: failed to read measurement values!
15:23:51.061 -> ICP20100: failed to read measurement values!
15:23:51.560 -> Most recent Pressure: 0, Temperature: 1.96
                                    ...
15:24:13.634 -> Most recent Pressure: 0, Temperature: 1.97
15:24:14.150 -> Most recent Pressure: 0, Temperature: 23.61
15:24:14.652 -> Most recent Pressure: 0, Temperature: 23.61

**Comment by Kriechi** [comment]
@jenyichen the last temperature values of 23.61 look reasonable - so the general connection seems to work at least partially.
Do you have another sensor chip to test with? maybe during soldering it got damaged?

I'm also testing with an ESP32 - so it should work. Did you set any special settings? I2C clock? CPU frequency?

**Comment by jenyichen** [comment]
Hi Kriechi,
Thank you for your reply. I had tried 3 pcs of the barometers and got the same issues. Another finding is once the temperature reading is non-zero, the temperature reading will not change even my finger touch it. I connected the VDD, VDDIO, CSB, and SDO together to 3.3V, so the slave address is 0x64. Software settings use default values in Arduino IDE 2.0.3.

**Comment by Kriechi** [comment]
Please try changing the sample code to print the 0th element instead:

original sample:
`Serial.printf("Most recent Pressure: %d, Temperature: %0.2f\n", pressure_values[count], temperature_values[count]);`

with new change:
`Serial.printf("Most recent Pressure: %d, Temperature: %0.2f - out of %d values\n", pressure_values[0], temperature_values[0], count);`

**Comment by jenyichen** [comment]
Now the temperature reading will change with ambient conditions!

09:16:04.318 -> ICP20100: failed to read measurement values!
09:16:04.818 -> ICP20100: failed to read measurement values!
09:16:05.317 -> Most recent Pressure: 0, Temperature: 25.89 - out of 1 values
09:16:05.816 -> Most recent Pressure: 0, Temperature: 25.83 - out of 1 values
09:16:06.317 -> Most recent Pressure: 0, Temperature: 25.83 - out of 1 values
09:16:06.817 -> Most recent Pressure: 0, Temperature: 25.83 - out of 1 values
09:16:07.318 -> Most recent Pressure: 0, Temperature: 25.70 - out of 1 values
09:16:07.831 -> Most recent Pressure: 0, Temperature: 25.76 - out of 1 values
09:16:08.332 -> Most recent Pressure: 0, Temperature: 25.89 - out of 1 values
09:16:08.814 -> Most recent Pressure: 0, Temperature: 27.03 - out of 1 values
09:16:15.317 -> Most recent Pressure: 0, Temperature: 28.87 - out of 1 values
09:16:15.865 -> Most recent Pressure: 0, Temperature: 28.75 - out of 1 values
09:16:16.363 -> Most recent Pressure: 0, Temperature: 28.68 - out of 1 values
09:16:16.863 -> Most recent Pressure: 0, Temperature: 28.62 - out of 1 values
09:16:17.362 -> Most recent Pressure: 0, Temperature: 28.62 - out of 1 values
09:16:17.861 -> Most recent Pressure: 0, Temperature: 28.49 - out of 1 values

**Comment by Kriechi** [comment]
Please try this example code:
```
#include <Arduino.h>
#include <ICP20100.h>

ICP20100 icp20100 = ICP20100();

void setup()
{
  delay(2000); // let everything fully boot and settle down
  Serial.begin(115200);
  Wire.begin();

  if (!icp20100.begin())
  {
    Serial.println("ICP20100 not detected or failed to initialize!");
    while (1) { ;; }
  }
}

void loop()
{
  delay(1000); // let the measurement sample accumulate in the sensor

  uint8_t count;
  float pressure_values[16];    // there are up to 16 samples to be read
  float temperature_values[16]; // there are up to 16 samples to be read
  if (!icp20100.read(&count, pressure_values, temperature_values))
  {
    Serial.println("ICP20100: no samples available or read failure!");
    delay(1000);
    return;
  }
  if (count > 0)
  {
    Serial.printf("Received %d samples - most recent Pressure: %f kPa, Temperature: %0.2f Celsius\n", count, pressure_values[count-1], temperature_values[count-1]);
  }
}
```

**Comment by jenyichen** [comment]
Great, it works fine.

09:23:00.351 -> ICP20100: no samples available or read failure!
09:23:02.373 -> ICP20100: no samples available or read failure!
09:23:04.383 -> Received 4 samples - most recent Pressure: 101.094971 kPa, Temperature: 23.03 Celsius
09:23:05.400 -> Received 2 samples - most recent Pressure: 101.095276 kPa, Temperature: 23.10 Celsius
09:23:06.387 -> Received 2 samples - most recent Pressure: 101.094360 kPa, Temperature: 23.03 Celsius
09:23:07.403 -> Received 2 samples - most recent Pressure: 101.093750 kPa, Temperature: 23.16 Celsius

**Comment by Kriechi** [comment]
Fixed by 5b23bd9.
